docs/challenges/language-support/csharp.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: "C#"
+description: "Learn how to solve CodeCrafters challenges in C#"
+---
+
+# Local development
+
+To run the code for your challenge locally, you'll need `dotnet` installed. Our test runners use Dotnet 8.0 (as of April 2024).
+
+The script for your challenge (like `./your_sqlite3.sh`) will automatically compile your code using `dotnet` before executing it. It'll
+look something like this:
+
+```sh
+#!/bin/sh
+
+exec dotnet run --project . --configuration Release "$@"
+```
+
+# File structure
+
+- The files for your solution are placed in the `src/` directory.
+- `src/Program.cs` contains the `main` function, which is what the test runner executes.
+- The `.csproj` file defines project settings, target frameworks and dependencies.
+
+# Adding more files
+
+You can add more files and directories to `src/`. The test runner will include them when compiling your code.
+
+For example, if you added a file at `src/Foo.cs`, you could use it like so:
+
+<CodeGroup>
+```csharp src/Foo.cs
+public class Foo {
+    public static void foo() {
+        System.out.println("Hello world!");
+    }
+}
+```
+
+```csharp src/Program.cs
+public class Main {
+    public static void Main(String[] args) {
+        Foo.foo();  // calls the method foo() from Foo class in Foo.cs
+    }
+}
+```
+
+</CodeGroup>
+
+By default, all classes in `src/` belong to the same namespace. This namespace is called the _project namespace_. E.g. If the project name is `MyProject`, all classes are defined under the `MyProject` namespace. This is why you can call `foo()` without importing anything.
+
+If you want to define a class under a _different_ namespace, you'll need to enclose it within braces like this:
```

Comment:
```suggestion
If you want to define a class under a _different_ namespace, you'll need to use the [`namespace`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/namespace) keyword like this:
```

(More important to highlight the `namespace` keyword than the braces)

Reply:
![CleanShot 2024-04-23 at 18 35 40@2x](https://github.com/codecrafters-io/docs/assets/71697384/f4e54c6f-e26c-4d39-ac05-f5f0a6f303df)
@rohitpaulk Hyperlinking a code block makes the styling look wonky. Better to remove the code blocks around namespace?